python/pyarrow/acero.py

```diff
@@ -53,6 +53,13 @@ class InMemoryDataset:
 def _dataset_to_decl(dataset, use_threads=True):
     decl = Declaration("scan", ScanNodeOptions(dataset, use_threads=use_threads))
 
+    # Get rid of special dataset columns
+    # "__fragment_index", "__batch_index", "__last_in_fragment", "__filename"
+    projections = [field(f) for f in dataset.schema.names]
+    decl = Declaration.from_sequence(
+        [decl, Declaration("project", ProjectNodeOptions(projections))]
+    )
```

Comment:
@westonpace do you know if there is an easier way to do this? (there is not an option to just turn off adding those fields in the scanner to start with?)

Reply:
This was on my "new scan node" todo list but sadly that list has been stalled for a while now.